### ISSUE TITLE
Can't receive signals when register handlers by `connect_via`

### ISSUE THREAD

**Issue by guyskk** [pallets/flask-sqlalchemy#505]
`@models_committed.connect_via(app)` not work:

```python
@models_committed.connect_via(app)
def on_models_committed(sender, changes):
    print(changes)
```
but `@models_committed.connect` works.

I spent a few hours to find out the reason, reproduce:

```python
# app.py
from flask import Flask
from flask_sqlalchemy import SQLAlchemy, models_committed

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///:memory:'
app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = True
app.config['DEBUG'] = True

print('create_app', type(app), id(app))
db = SQLAlchemy(app)


class User(db.Model):
    id = db.Column(db.Integer, primary_key=True)


@models_committed.connect_via(app)
def on_models_committed(sender, changes):
    print(changes)


@app.route('/')
def index():
    user = User()
    db.session.add(user)
    db.session.commit()
    return 'User id=' + str(user.id)


if __name__ == '__main__':
    db.create_all()
    with app.test_client() as client:
        print(client.get('/').data)
```

And add `print('flask_sqlalchemy', type(session.app), id(session.app))` to `flask_sqlqlchemy/__init__.py` for debuging:

```
    @staticmethod
    def after_commit(session):
        try:
            d = session._model_changes
        except AttributeError:
            return

        if d:
            print('flask_sqlalchemy', type(session.app), id(session.app))
            models_committed.send(session.app, changes=list(d.values()))
            d.clear()
```

Run the scripts:
```bash
$ python app.py
create_app <class 'flask.app.Flask'> 140350331860528
flask_sqlalchemy <class 'werkzeug.local.LocalProxy'> 140350227515936
b'User id=1'
```

`session.app` is not `app`, so `connect_via` not work.

**Comment by guyskk** [comment]
Another strange thing, `models_committed.connect` in `create_app` not working:

```python
from flask import Flask
from flask_sqlalchemy import SQLAlchemy, models_committed

db = SQLAlchemy()


class User(db.Model):
    id = db.Column(db.Integer, primary_key=True)


# == works here ==
@models_committed.connect
def on_models_committed_1(sender, changes):
    print('on_models_committed_1')
    print(changes)


def create_app():
    app = Flask(__name__)
    app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///:memory:'
    app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = True
    app.config['DEBUG'] = True
    print('create_app', type(app), id(app))

    # == Not working here ==
    @models_committed.connect
    def on_models_committed_2(sender, changes):
        print('on_models_committed_2')
        print(changes)

    db.init_app(app)

    @app.route('/')
    def index():
        user = User()
        db.session.add(user)
        db.session.commit()
        return 'User id=' + str(user.id)

    return app


if __name__ == '__main__':
    app = create_app()
    db.create_all(app=app)
    with app.test_client() as client:
        print(client.get('/').data)
```

outputs:
```bash
$ python app2.py
create_app <class 'flask.app.Flask'> 139956199525512
flask_sqlalchemy <class 'werkzeug.local.LocalProxy'> 139956174953208
on_models_committed_1
[(<__main__.User object at 0x7f4a145c87f0>, 'insert')]
b'User id=1'
```
Don't know why :(

**Comment by adimux** [comment]
For the second issue, it could be because the signal holds a weak reference to the receiver (ref: https://pythonhosted.org/blinker/) and since it's a local function, it's getting destroyed by python's garbage collector. Try using `weak=False`.

**Comment by davidism** [comment]
Fixed by #547.
